packages/core/nuxt-theme-module/theme/helpers/ui/index.ts

```diff
@@ -0,0 +1,78 @@
+
```

Comment:
these fns definitely need descriptions on how to be implemented and what they do (ideally a document in docs so warns can just link to it) and should be in the boilerplate but lets leave it for our discussion

Reply:
NOTE: replace factory to object